components/event-processor-manager/org.wso2.carbon.event.processor.manager.commons/src/main/java/org/wso2/carbon/event/processor/manager/commons/transport/client/TCPEventPublisher.java

```diff
@@ -277,6 +292,11 @@ private void disconnect() {
             }
         } catch (IOException e) {
             log.debug("Error while closing socket to " + hostUrl + " : " + e.getMessage(), e);
+        }  finally {
+            if(connectionCallback != null){
+                connectionCallback.onCepReceiverDisconnect();
+                log.info("--------------------------TCPEventPublisher onclose invoked.");//todo: delete later
```

Comment:
oops! I searched & deleted all the comments but somehow have missed this.. :(
Deleted now!